src/neighbors.cc

```diff
@@ -1269,54 +1257,12 @@ peerRefreshDNS(void *data)
     }
 
     for (p = Config.peers; p; p = p->next)
-        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p);
+        lookupPeer(p);
 
     /* Reconfigure the peers every hour */
     eventAddIsh("peerRefreshDNS", peerRefreshDNS, nullptr, 3600.0, 1);
 }
 
-// TODO: Move to CachePeer::noteFailure() or similar.
-// TODO: Require callers to detail failures instead of using one (and often
-// misleading!) "TCP" label for all of them.
-void
-peerConnectFailed(CachePeer * const p)
-{
-    p->stats.last_connect_failure = squid_curtime;
-    if (p->tcp_up > 0)
-        --p->tcp_up;
-
-    // TODO: Report peer name. Same-addresses peers often have different names.
-
-    const auto consideredAliveByAdmin = p->stats.logged_state == PEER_ALIVE;
-    const auto level = consideredAliveByAdmin ? DBG_IMPORTANT : 2;
-    debugs(15, level, "ERROR: TCP connection to " << p->host << "/" << p->http_port << " failed");
-
-    if (consideredAliveByAdmin) {
-        if (!p->tcp_up) {
-            debugs(15, DBG_IMPORTANT, "Detected DEAD " << neighborTypeStr(p) << ": " << p->name);
-            p->stats.logged_state = PEER_DEAD;
-        } else {
-            debugs(15, 2, "additional failures needed to mark this cache_peer DEAD: " << p->tcp_up);
-        }
-    } else {
-        assert(!p->tcp_up);
-        debugs(15, 2, "cache_peer " << p->host << "/" << p->http_port << " is still DEAD");
-    }
-}
-
-void
-peerConnectSucceded(CachePeer * p)
-{
-    if (!p->tcp_up) {
-        debugs(15, 2, "TCP connection to " << p->host << "/" << p->http_port << " succeeded");
-        p->tcp_up = p->connect_fail_limit; // NP: so peerAlive(p) works properly.
-        peerAlive(p);
-        if (!p->n_addresses)
-            ipcache_nbgethostbyname(p->host, peerDNSConfigure, p);
```

Comment:
I think we can move this `if` w/o side effects because other two callers, `neighborAlive()` and `neighborAliveHtcp()` are invoked only if n_addresses>0.

Reply:
OK, let's move it because that would reduce the amount of other code you have to move/name in this PR.